include/mbedtls/serialize.h

```diff
@@ -0,0 +1,195 @@
+/**
+ * \file serialize.h
+ *
+ * \brief Serializer for operating system functions
+ *
+ * \warning THIS IS AN UNSTABLE INTERFACE!
+ *  The interfaces in this module are internal to the mbed TLS project. They
+ *  are meant for testing purposes only. They may change at any time.
+ *
+ *  Copyright (C) 2017, ARM Limited, All Rights Reserved
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  This file is part of mbed TLS (https://tls.mbed.org)
+ */
+
+#ifndef MBEDTLS_SERIALIZE_H
+#define MBEDTLS_SERIALIZE_H
+
+#if !defined(MBEDTLS_CONFIG_FILE)
+#include "config.h"
+#else
+#include MBEDTLS_CONFIG_FILE
+#endif
+
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+/** Maximum length of a serialized parameter. */
+#define MBEDTLS_SERIALIZE_MAX_STRING_LENGTH 0xffffff
+
+/** Offloaded function identifiers */
+#define MBEDTLS_SERIALIZE_TYPE_PUSH     0x50 /**< T>H Push input parameter */
+#define MBEDTLS_SERIALIZE_TYPE_EXECUTE  0x45 /**< T>H Execute function */
+#define MBEDTLS_SERIALIZE_TYPE_RESULT   0x72 /**< H>T Output parameter */
+
+#define MBEDTLS_SERIALIZE_FUNCTION_EXIT         0x000100 /**< exit the frontend */
```

Comment:
It would be good to explain the encoding of the number of output and input parameters as bits [0:3] and [4:7] here.

Reply:
@mazimkhan Could you please add a reference to `library/serialize.c` here, so that one can look up the encoding of the identifiers?